endurance-page-cache.php

```diff
@@ -972,37 +972,59 @@ public function uri_to_cache( $uri ) {
 		 *
 		 * @return bool
 		 */
-		public function is_cachable() {
+		public function is_cachable( $type = 'default' ) {
 			global $wp_query;
 
 			$return = true;
 
-			if ( defined( 'DONOTCACHEPAGE' ) && DONOTCACHEPAGE === true ) {
-				$return = false;
-			} elseif ( defined( 'DOING_AJAX' ) ) {
-				$return = false;
-			} elseif ( 'private' === get_post_status() ) {
-				$return = false;
-			} elseif ( isset( $wp_query ) && is_404() ) {
-				$return = false;
-			} elseif ( is_admin() ) {
-				$return = false;
-			} elseif ( false === get_option( 'permalink_structure' ) ) {
-				$return = false;
-			} elseif ( function_exists( 'is_user_logged_in' ) && is_user_logged_in() ) {
-				$return = false;
-			} elseif ( isset( $_GET ) && ! empty( $_GET ) ) { // phpcs:ignore WordPress.Security.NonceVerification
-				$return = false;
-			} elseif ( isset( $_POST ) && ! empty( $_POST ) ) { // phpcs:ignore WordPress.Security.NonceVerification
-				$return = false;
-			} elseif ( isset( $wp_query ) && is_feed() ) {
-				$return = false;
+			if ( 'file' === $type ) {
+				if ( defined( 'DONOTCACHEPAGE' ) && DONOTCACHEPAGE === true ) {
+					$return = false;
+				} elseif ( defined( 'DOING_AJAX' ) ) {
+					$return = false;
+				} elseif ( 'private' === get_post_status() ) {
+					$return = false;
+				} elseif ( isset( $wp_query ) && is_404() ) {
+					$return = false;
+				} elseif ( is_admin() ) {
+					$return = false;
+				} elseif ( false === get_option( 'permalink_structure' ) ) {
+					$return = false;
+				} elseif ( function_exists( 'is_user_logged_in' ) && is_user_logged_in() ) {
+					$return = false;
+				} elseif ( isset( $_GET ) && ! empty( $_GET ) ) { // phpcs:ignore WordPress.Security.NonceVerification
+					$return = false;
+				} elseif ( isset( $_POST ) && ! empty( $_POST ) ) { // phpcs:ignore WordPress.Security.NonceVerification
+					$return = false;
+				} elseif ( isset( $wp_query ) && is_feed() ) {
+					$return = false;
+				}
+				$cache_exempt = array_merge( $this->cache_exempt, array( '@', '%', ':', ';', '&', '=', '.' ) );
```

Comment:
Could define this up on line 52 with the other exempt list. cache_exempt and file_cache_exempt so it is easier to find.